src/Education/XamlGenerator/CustomTextMarkupParser.cs

```diff
@@ -98,6 +98,11 @@ internal class SimpleText : ISimpleText
 
         public SimpleText(string text)
         {
+            if (string.IsNullOrEmpty(text))
+            {
+                throw new ArgumentNullException("text");
```

Comment:
Use `nameof` where possible, rather the name of the argument as a string.


Also, there is a short-form for this using `??`:

```
Text = text ?? throw new ArgumentNullException(nameof(text))
```